analyzers/tests/SonarAnalyzer.UnitTest/Helpers/MethodParameterLookupTest.cs

```diff
@@ -25,287 +25,374 @@
+    }
 
-            protected InspectionBase(string source, AnalyzerLanguage language)
+    [TestMethod]
+    public void TestMethodParameterLookup_CS_MultipleCandidates()
+    {
+        var source = """
+            namespace Test
             {
-                Compiler = new SnippetCompiler(source, false, language);
-                MainInvocations = FindInvocationsIn("Main");
-            }
+                class TestClass
+                {
+                    void Main()
+                    {
+                        dynamic d = 42;
+                        AmbiguousCall(d);
+                    }
+                
+                    void AmbiguousCall(int p) { }
 
-            public MethodParameterLookupBase<TArgumentSyntax> CreateLookup(int invocationIndex, string expectedMethod)
-            {
-                var invocation = MainInvocations[invocationIndex];
-                var method = Compiler.SemanticModel.GetSymbolInfo(invocation).Symbol as IMethodSymbol;
-                method.Name.Should().Be(expectedMethod);
-                return CreateLookup(invocation, method);
+                    void AmbiguousCall(string p) { }
+                }
             }
+            """;
+        var compiler = new SnippetCompiler(source, false, AnalyzerLanguage.CSharp);
+        var lookup = new CSharpMethodParameterLookup(compiler.GetNodes<CSharpSyntax.InvocationExpressionSyntax>().Single(), compiler.SemanticModel);
 
-            public void CheckExpectedParameterMappings(int invocationIndex, string expectedMethod, object expectedArguments)
-            {
-                var lookup = CreateLookup(invocationIndex, expectedMethod);
-                InspectTryGetSyntax(lookup, expectedArguments, lookup.MethodSymbol);
-                InspectTryGetSymbol(lookup, expectedArguments, GetArguments(MainInvocations[invocationIndex]));
-            }
+        lookup.TryGetSyntax("p", out var expressions).Should().BeTrue();
+        expressions.Should().BeEquivalentTo(new { Identifier = new { ValueText = "d" } });
+    }
 
-            protected void InitSpecial(TInvocationSyntax specialInvocation)
-            {
-                SpecialArgument = GetArguments(specialInvocation).Single();
-                SpecialParameter = (Compiler.SemanticModel.GetSymbolInfo(specialInvocation).Symbol as IMethodSymbol).Parameters.Single();
-            }
+    [TestMethod]
+    public void TestMethodParameterLookup_VB_MultipleCandidates()
+    {
+        var source = """
+            Module Test
+                Sub Main()
+                    Overloaded(42, "")
+                End Sub
+
+                Sub Overloaded(a As Integer, b As Boolean)
+                End Sub
+
+                Sub Overloaded(a As Integer, b As Integer)
+                End Sub
+            End Module
+            """;
+        var compiler = new SnippetCompiler(source, true, AnalyzerLanguage.VisualBasic);
```

Comment:
Here too

Reply:
This suggestion is controversial in the team:
```suggestion
        var compiler = new SnippetCompiler(source, ignoreErrors: true, AnalyzerLanguage.VisualBasic);
```
What do you think about using named arguments for literals (true/false and sometimes numbers) when it is not obvious what that argument is about?

Reply:
I usually do that, but forgot to do it in this case. Helps a lot with readability.